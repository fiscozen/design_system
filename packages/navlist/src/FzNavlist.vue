<script lang="ts" setup>
import { FzNavlink } from '@fiscozen/navlink'
import { FzCollapse } from '@fiscozen/collapse'
import { FzNavlistProps, isSubMenu } from './types'

defineProps<FzNavlistProps>()
</script>

<template>
  <div class="fz__navlist inline-flex grow-0 flex-col rounded p-4">
    <div
      class="fz__navlist__section border-grey-200"
      v-for="(section, sectionIndex) in sections"
      :key="sectionIndex"
    >
      <div class="text-grey-400 flex h-32 items-center px-12 text-xs">
        <span>{{ section.label }}</span>
      </div>
      <div class="flex flex-col" v-for="(item, itemIndex) in section.items" :key="itemIndex">
        <FzCollapse v-if="isSubMenu(item)" :summary-class="'px-12'">
          <template #summary
            ><span class="grow">{{ item.summary }}</span></template
          >
          <template #content>
            <div class="flex flex-col">
              <FzNavlink
                class="grow-1 flex justify-start pl-24"
                v-for="(subitem, subIndex) in item.subitems"
                :key="subIndex"
                v-bind="subitem"
                >{{ subitem.label }}</FzNavlink
              >
            </div>
          </template>
        </FzCollapse>
        <FzNavlink class="grow-1 flex justify-start" v-else v-bind="item">{{
          item.label
        }}</FzNavlink>
      </div>
      <hr
        v-if="sectionIndex !== sections.length - 1"
        class="bg-grey-200 my-8 h-1 w-full border-none px-4"
      />
    </div>
  </div>
</template>

<style>
.fz__navlist {
  min-width: 240px;
  box-shadow:
    0px 1px 3px 0px rgba(0, 0, 0, 0.1),
    0px 1px 2px 0px rgba(0, 0, 0, 0.06);
}
</style>
