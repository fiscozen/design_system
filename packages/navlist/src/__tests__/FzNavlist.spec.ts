import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import FzNavlist from '../FzNavlist.vue'
const sections = [
  {
    label: 'Label 1',
    items: [
      {
        label: 'Item #1',
        meta: {
          path: '/foo',
          name: 'foo'
        }
      },
      {
        summary: 'Item #2',
        subitems: [
          {
            label: 'Sub-Item #1',
            meta: {
              path: '/foo',
              name: 'foo'
            }
          },
          {
            label: 'Sub-Item #2',
            meta: {
              path: '/foo',
              name: 'foo'
            }
          }
        ]
      }
    ]
  },
  {
    label: 'Label 2',
    items: [
      {
        label: 'Item #1',
        disabled: true,
        meta: {
          path: '/foo',
          name: 'foo'
        }
      },
      {
        label: 'Item #2',
        meta: {
          path: '/foo',
          name: 'foo'
        }
      }
    ]
  }
]
describe('FzNavlist', () => {
  it('should match snapshot', () => {
    const wrapper = mount(FzNavlist, {
      props: {
        sections
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
