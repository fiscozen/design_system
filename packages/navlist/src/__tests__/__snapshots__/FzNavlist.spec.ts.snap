// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzNavlist > should match snapshot 1`] = `
"<div class="fz__navlist inline-flex grow-0 flex-col rounded p-4">
  <div class="fz__navlist__section border-grey-200">
    <div class="text-grey-400 flex h-32 items-center px-12 text-xs"><span>Label 1</span></div>
    <div class="flex flex-col"><button class="px-12 py-6 text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent font-medium grow-1 flex justify-start">
        <!--v-if-->Item #1
      </button></div>
    <div class="flex flex-col">
      <details data-v-5d0ac24e="" data-e2e="details">
        <summary data-v-5d0ac24e="" data-e2e="summary" class="text-grey-500 flex h-32 cursor-pointer select-none list-none items-center text-sm rounded font-medium px-12"><span class="grow">Item #2</span>
          <div data-v-5d0ac24e="" class="flex items-center justify-center w-[20px] h-[20px] ml-32"><svg class="svg-inline--fa fa-chevron-down h-[16px]" aria-hidden="true" focusable="false" data-prefix="far" data-icon="chevron-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
              <path class="" fill="currentColor" d="M239 401c9.4 9.4 24.6 9.4 33.9 0L465 209c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0l-175 175L81 175c-9.4-9.4-24.6-9.4-33.9 0s-9.4 24.6 0 33.9L239 401z"></path>
            </svg></div>
        </summary>
        <transition-stub data-v-5d0ac24e="" appear="false" persisted="true" css="true">
          <div data-v-5d0ac24e="" data-e2e="content" class="text-sm" style="display: none;">
            <div class="flex flex-col"><button class="px-12 py-6 text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent font-medium grow-1 flex justify-start pl-24 !font-normal">
                <!--v-if-->Sub-Item #1
              </button><button class="px-12 py-6 text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent font-medium grow-1 flex justify-start pl-24 !font-normal">
                <!--v-if-->Sub-Item #2
              </button></div>
          </div>
        </transition-stub>
      </details>
    </div>
    <hr class="bg-grey-200 my-8 h-1 w-full border-none px-4">
  </div>
  <div class="fz__navlist__section border-grey-200">
    <div class="text-grey-400 flex h-32 items-center px-12 text-xs"><span>Label 2</span></div>
    <div class="flex flex-col"><button disabled="" class="px-12 py-6 text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent font-medium grow-1 flex justify-start">
        <!--v-if-->Item #1
      </button></div>
    <div class="flex flex-col"><button class="px-12 py-6 text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent font-medium grow-1 flex justify-start">
        <!--v-if-->Item #2
      </button></div>
    <!--v-if-->
  </div>
</div>"
`;
